python/dgl/data/utils.py

```diff
@@ -350,3 +361,395 @@ def __len__(self):
+
+    There can be multiple NLP encoders in nlps. Each encoder
+    is invoded to generate a embedding for the input string and
+    the resulting embeddings are concatenated.
+
+    Parameters
+    ----------
+    str_val : str
+        words to encode
+
+    nlps : list of func
+        a list of nlp encoder functions
+    """
+    vector = None
+    for nlp in nlps:
+        doc = nlp(str_val)
+        if vector is None:
+            vector = doc.vector
+        else:
+            vector = np.concatenate((vector, doc.vector))
+    return vector
+
+def parse_lang_feat(str_feats, nlps, verbose=False):
+    """ Parse a list of strings using word2vec encoding using NLP encoders in nlps
+
+    Parameters
+    ----------
+    str_feats : list of str
+        list of strings to encode
+
+    nlps : list of func
+        a list of nlp encoder functions
+
+    verbose : bool, optional
+        print out debug info
+        Default: False
+
+    Return
+    ------
+    numpy.array
+        the encoded features
+    """
+    features = []
+    num_feats = len(str_feats)
+    num_process = num_feats if num_feats < 8 else 8 # TODO(xiangsx) get system nproc
+    batch_size = (num_feats + num_process - 1) // num_process
+
+    def embed_lang(d, proc_idx, feats):
+        res_feats = []
+        for s_feat in feats:
+            res_feats.append(embed_word2vec(s_feat, nlps))
+        d[proc_idx] = res_feats
+
+    # use multi process to process the feature
+    manager = Manager()
+    d = manager.dict()
+    job=[]
+    for i in range(num_process):
+        sub_info = str_feats[i * batch_size : (i+1) * batch_size \
+                         if (i+1) * batch_size < num_feats else num_feats]
+        job.append(Process(target=embed_lang, args=(d, i, sub_info)))
+
+    for p in job:
+        p.start()
+
+    for p in job:
+        p.join()
+
+    for i in range(num_process):
+        if len(d[i]) > 0:
+            features.append(d[i])
+
+    features = np.concatenate(features)
+    if verbose:
+        print(features.shape)
+
+    return features
+
+def parse_word2vec_node_feature(str_feats, langs, verbose=False):
+    """ Parse a list of strings using word2vec encoding using NLP encoders in nlps
+
+    Parameters
+    ----------
+    str_feats : list of str
+        list of strings to encode
+
+    nlps : list of func
+        a list of nlp encoder functions
+
+    verbose : bool, optional
+        print out debug info
+        Default: False
+
+    Return
+    ------
+    numpy.array
+        the encoded features
+
+    Examples
+    --------
+
+    >>> inputs = ['hello', 'world']
+    >>> languages = ['en_core_web_lg', 'fr_core_news_lg']
+    >>> feats = parse_word2vec_node_feature(inputs, languages)
+
+    """
+    import spacy
+
+    nlps = []
+    for lang in langs:
+        nlp = spacy.load(lang)
+        nlps.append(nlp)
+
+    return parse_lang_feat(str_feats, nlps, verbose)
+
+def parse_category_single_feat(category_inputs, norm=None):
```

Comment:
Reference in pandas https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.get_dummies.html